src/messaging/ExchangeContext.cpp

```diff
@@ -239,14 +239,28 @@ CHIP_ERROR ExchangeContext::SendMessage(Protocols::Id protocolId, uint8_t msgTyp
             }
         }
 
-        // Standalone acks are not application-level message sends.
-        if (!isStandaloneAck && err == CHIP_NO_ERROR)
+        if (err == CHIP_NO_ERROR)
         {
-            //
-            // Once we've sent the message successfully, we can clear out the WillSendMessage flag.
-            //
-            mFlags.Clear(Flags::kFlagWillSendMessage);
-            MessageHandled();
+#if CONFIG_DEVICE_LAYER && CHIP_CONFIG_ENABLE_ICD_SERVER
+            DeviceLayer::ChipDeviceEvent event;
+            event.Type                       = DeviceLayer::DeviceEventType::kChipMsgSentEvent;
+            event.MessageSent.ExpectResponse = IsResponseExpected();
```

Comment:
So this is going to kick us into active mode for a bit for sending:

1. Non-resend messages that expect a response.
2. Non-resend messages that don't expect a response.
3. Standalone acks.

But will not  kick us into active mode when we do an MRP retransmit of a message.

Is that the intended behavior?  If so, why is it the intended behavior?  I would really like to understand what part of the spec this is implementing and why we think the above description is the right implementation.

Reply:
As discussed, I opened to a ticket for me to track and add the logic/event posting in the MRP retransmit.
https://github.com/project-chip/connectedhomeip/issues/27864